fix(sf|random): ST_GENERATEPOINTS was not accepting column names

vdelacruzb opened the pull request:
# Description

Shortcut

- Story: https://app.shortcut.com/cartoteam/story/386028/bug-report-st-generatepoints-function-not-accepting-column-names
- Autolink: [sc-386028]

Updated the implementation to make if full Javascript. It's 3x faster and ensures to get the exact number of points.
On the other side I'm returning the JSON.stringified version as it was working before. Noticed that it wouldn't be needed but kept it as it was just in case.

## Type of change

- Fix

# Acceptance

```
CREATE or replace TABLE CARTO_DATA_ENGINEERING_TEAM.vdelacruz_carto.test_random_points AS
SELECT TO_GEOGRAPHY('POLYGON((0 0, 10 0, 10 10, 0 0))') as geog, 1230 as npoints
UNION ALL
SELECT TO_GEOGRAPHY('POLYGON((0 0, -10 0, -10 -10, 0 0))') as geog, 2340 as npoints;

with _a AS (
SELECT CARTO_DEV_DATA.CARTO.ST_GENERATEPOINTS(TO_GEOGRAPHY('POLYGON((0 0, 10 0, 10 10, 0 0))'), 1000) arr
)
SELECT ST_NPOINTS(TO_GEOGRAPHY(element.VALUE)) FROM a, LATERAL FLATTEN(input=>arr) as element
;
-- OK 628 ms

WITH _a AS (
SELECT CARTO_DEV_DATA.CARTO.ST_GENERATEPOINTS(geog, npoints) arr
FROM CARTO_DATA_ENGINEERING_TEAM.vdelacruz_carto.test_random_points
)
SELECT ST_NPOINTS(TO_GEOGRAPHY(element.VALUE)) FROM _a, LATERAL FLATTEN(input=>arr) as element
;
-- Unsupported subquery type cannot be evaluated

with _a AS (
SELECT CARTO_DATA_ENGINEERING_TEAM.vdelacruz_carto.ST_GENERATEPOINTS(TO_GEOGRAPHY('POLYGON((0 0, 10 0, 10 10, 0 0))'), 1000) arr
)
SELECT ST_NPOINTS(TO_GEOGRAPHY(element.VALUE)) FROM _a, LATERAL FLATTEN(input=>arr) as element
;
-- OK 137 ms

WITH _a AS (
SELECT CARTO_DATA_ENGINEERING_TEAM.vdelacruz_carto.ST_GENERATEPOINTS(geog, npoints) arr
FROM CARTO_DATA_ENGINEERING_TEAM.vdelacruz_carto.test_random_points
)
SELECT ST_NPOINTS(TO_GEOGRAPHY(element.VALUE)) FROM _a, LATERAL FLATTEN(input=>arr) as element
;
-- OK 361 ms
```

Jesus89 reviewed (commented):
LGTM, after fixing CI

vdelacruzb commented:
> LGTM, after fixing CI

It seems that `processing` tests were malfunctioning, just fixed them. Also updated the test to use a table in `ST_GENERATEPOINTS`.
